.github/workflows/cache_data.yaml

```diff
@@ -74,3 +77,5 @@ jobs:
           path: |
               ~/.gmt/cache
               ~/.gmt/server
+              ~/.gmt/gmt_data_server.txt
+              ~/.gmt/gmt_hash_server.txt
```

Comment:
Aren't these files supposed to be downloaded automatically?

Reply:
Ideally yes, but old GMT versions may fail to download these files on GitHub Actions (see https://github.com/GenericMappingTools/pygmt/actions/runs/7619206138/job/20751859452):
```
Error: ession [ERROR]: Read error first record in file /home/runner/.gmt/server/gmt_data_server.txt
Error: ession [ERROR]: Deleting /home/runner/.gmt/server/gmt_data_server.txt so it can get regenerated - please try again
Error: h [ERROR]: Unable to open file /home/runner/.gmt/server/gmt_data_server.txt
Error: h [ERROR]: Libcurl Error: HTTP response code said error
Warning: [WARNING]: You can turn remote file download off by setting GMT_DATA_UPDATE_INTERVAL to "off"
Error: h [ERROR]: File earth_relief_01d_p not found!
```

Reply:
Ah got it, so this is needed for the GMT legacy tests (using GMT 6.3.0 and 6.4.0). I see that the path has changed from `~/.gmt/server/gmt_data_server.txt` to `~/.gmt/gmt_data_server.txt`

Reply:
Do we need to copy `~/.gmt/gmt_data_server.txt` to `~/.gmt/server/gmt_data_server.txt` though for the GMT Legacy tests?

Reply:
> Do we need to copy `~/.gmt/gmt_data_server.txt` to `~/.gmt/server/gmt_data_server.txt` though for the GMT Legacy tests?

My local tests tell me that we have to. Will do it now.

Reply:
Done in 49178fcb4717dc7a6b63066357a1b048499b4969 and the same error is gone in the GMT Legacy Tests workflow (https://github.com/GenericMappingTools/pygmt/actions/runs/7677658040/job/20926661680?pr=3020).